Run TMB and mutations plots(5/N)

AntoniaChroni opened the pull request:
<!--Hi there, thanks for your contribution! Please take a moment to fill out this template to facilitate the review of your pull request.-->

### Purpose/implementation Section

#### What scientific question is your analysis addressing?

We explored TMB and the number of mutations across longitudinal samples with genomic assays. 

Generally, Hypermutant TMB defined as ≤10 Mb, and Ultrahypermutant TMB defined as >100 mutations/Mb (https://pubmed.ncbi.nlm.nih.gov/29056344/). There are hyper mutant samples in the PBTA cohort. We excluded samples with >= 10 from downstream analysis. Attention is needed in cases with high number of mutations in only one timepoint as this will lead to un-matched longitudinal samples. We also remove those so we always have matched longitudinal samples.


#### What was your approach?



#### What GitHub issue does your pull request address?



### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

#### Which areas should receive a particularly close look?

Please review for script logic.

#### Is there anything that you want to discuss further?


#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?



### Results

#### What types of results are included (e.g., table, figure)?

There are 10 plots in the `plots_dir`. These are also summarized in the html report, which also contains more plots related to mutation count per `bs_id` and per `kids_id`.

#### What is your summary of the results?

Overall, TMB patterns across timepoints seem to be idiosyncratic, but we notice a higher overall TMB for Progressive and Recurrence compared to other timepoints. We don’t see any patterns driven by cancer type.

We also explored the total number of mutations per biospecimen sample and per timepoint in each patient case. This is to showcase the importance of the number of samples per timepoint and how each biopsy might be capturing a different number of mutations (and potentially of driver genes). 


#### Reproducibility Checklist

<!-- Check all those that apply or remove this section if it is not applicable.-->

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.

#### Documentation Checklist

<!-- Please review https://github.com/d3b-center/pbta-tumor-evolution#documenting-your-analysis -->

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

<!-- IF YOUR PULL REQUEST IS A DATA RELEASE, PLEASE REMOVE THE [HTML COMMENT TAG](https://html.com/tags/comment-tag/) FROM THE SECTION BELOW AND COMPLETE THE CHECKLIST-->

<!-- 
### Data Release Checklist
- [ ] Is the table in doc/data-file-descriptions.md up to date?
- [ ] Is doc/data-format.md up to date?
- [ ] Is doc/release-notes.md up to date?
- [x] Is download-data.sh up to date?
- [ ] Was download-data.sh tested and did it complete without error?
-->

Commit:
Add script, README, html, plots for explore-TMB-mut

Commit:
Edit README file about function-create-dumbbell-plot & function-create-barplot

Commit:
minor edits in README file

Commit:
Edit palette_df name

Commit:
Rename tmb_genomic name as df

Commit:
Rename plot_df to df_plot

Commit:
Replace ylim = max(df_plot_filter$tmb)

Commit:
Replace ylim for samples code chunk

Commit:
Add create-barplot-sample-panel code chunk

Commit:
Remove copies of the functions

Commit:
Edit name files for functions

Commit:
Merge branch 'achroni/explore-TMB-mut-functions' into achroni/explore-TMB-mut-script

Commit:
Merge branch 'achroni/explore-TMB-mut-functions' into achroni/explore-TMB-mut-script

Commit:
Add code chunk for adjusting ylim (draft)

Commit:
update plots

Commit:
Merge branch 'achroni/explore-TMB-mut-functions' into achroni/explore-TMB-mut-script

Commit:
Merge branch 'main' into achroni/explore-TMB-mut-script

Commit:
Fix typos

Commit:
Edit #patients with TMB info in README

Commit:
Select col in df, clarification in steps for filtering, add all timepoints for filtering in df_plot_filter

Commit:
Merge branch 'achroni/explore-TMB-mut-functions' into achroni/explore-TMB-mut-script

Commit:
Add jitter plot code chunk and explanation on what happens with missing values

Commit:
Fix sample-mutations-panel code chunk

Commit:
Merge branch 'achroni/explore-TMB-mut-functions' into achroni/explore-TMB-mut-script

Commit:
Rename dumbbell plots

Commit:
Remove word genomic from naming plots

Commit:
Fix comma typo

Commit:
Re-arrange code chunks order

Commit:
Run in Docker

Commit:
Add cg_sum code

Commit:
Add cg_plot code

Commit:
Remove ylim if statement and replace with max ylim for all cg_plot

Commit:
Define ylim based on cg_plot

Commit:
Change fig size for code chunk

Commit:
Remove code chunk for dumbell plots for #mutation count

Commit:
Remove code chunk for jitter plots

Commit:
Remove code chunk create-barplot-sample-panel

Commit:
Remove source jitter function

Commit:
Remove part for saving barplots per kids_id

Commit:
Merge branch 'achroni/explore-TMB-mut-functions' into achroni/explore-TMB-mut-script

Commit:
Run Rmd in Docker

Commit:
Add plots results

naqvia reviewed (changes requested):
The script logic makes sense to me and runs as expected. I do have some general suggestions, more related to plots: 

`TMB-genomic-no-hypermutants.pdf`: Text size for axis and legend titles can be larger
`TMB-genomic-total.pdf`: The sample labels are squished together and could be spaced out better. Also you can increase y-axis tick breaks (we can't really make out the smaller/ anything < 100 TMB).

rjcorb commented:
I think you want to merge this into `achroni/explore-TMB-mut-functions` so that the functions scripts aren't listed as new files.

rjcorb commented on analyses/tmb-vaf-longitudinal/03-explore-TMB-mut.Rmd:
```diff
@@ -0,0 +1,221 @@
+  p <- create_barplot_sample(tmb_df = tmb_sub,
+                             ylim = ylim,
+                             sid = samples[i])
+  print(p)
```
should these be saved as well?

AntoniaChroni replied:
We discussed this with JL and decided to only print them in the notebook and not save them in the directory.

rjcorb commented on analyses/tmb-vaf-longitudinal/03-explore-TMB-mut.Rmd:
```diff
@@ -0,0 +1,221 @@
+        count(cg_plot, Kids_First_Participant_ID))
+
+# Dumbbell plot per cancer group
+cancer_groups <- unique(as.character(df_plot_filter$cg_plot))
+cancer_groups <- sort(cancer_groups, decreasing = FALSE)
+print(cancer_groups)
```
How are patients with both low-grade and high-grade glioma diagnoses being handled here?

AntoniaChroni replied:
This is a great question. I am addressing this issue but identifying samples with different diagnosis across timepoints and summarizing them in the `cg_sum` col generated in the `read_input_files` code chunk. That means if there are cases with different diagnosis at timepoints, there will show as, e.g., `Low-grade glioma,High-grade glioma`. If it is the same diagnosis across timepoints it will show as e.g., `Low-grade glioma`. Plots are named accordingly.

Parts of the relevant code showing below:

```
%>% mutate(cg_distinct = n_distinct(cancer_group) > 1) # to identify samples with different diagnosis across timepoints
```
<more_code>

```
%>% summarise(cg_sum = str_c(cancer_group, collapse = ",")) %>% # to identify cases with multiple diagnosis
```

Here, I am using the `cg_plot` col in which I have summarized the samples with less than 10 bs as other and retaining the information from the `cg_sum` column. 


```
# Let's summarize cancer groups with < 10 bs_samples as Other and use this for visualization purposes
cg_sum_df <- df %>% 
  count(cg_sum) %>% 
  dplyr::mutate(cg_sum_n = glue::glue("{cg_sum} (N={n})"))

df <- df %>% 
  left_join(cg_sum_df, by = c("cg_sum")) %>% 
  mutate(cg_plot = case_when(n < 10 ~ "Other",
                             TRUE ~ cg_sum))

```


@rjcorb Please let me know if more clarification is needed.

rjcorb commented on analyses/tmb-vaf-longitudinal/03-explore-TMB-mut.Rmd:
```diff
@@ -0,0 +1,221 @@
+# Run function
+fname <- paste0(plots_dir, "/", "TMB-genomic-no-hypermutants.pdf")
+print(fname)
+p <- create_stacked_barplot(tmb_df = df_plot_filter, ylim = ylim)
```
This looks good! Is there any way you could group patients by cancer group here?

AntoniaChroni replied:
Done, see 504c0590334c6c8520e5698f01ec50f83711e4af

rjcorb commented on analyses/tmb-vaf-longitudinal/README.md:
```diff
@@ -15,7 +15,7 @@ bash run-tmb-vaf-longitudinal.sh
 
 ## Summary 
-This pipeline produces pdf files containing the correlation of VAFs (with label genes of interest), as well as violin plots and stacked barplots for TMB of paired longitudinal samples under various timepoints models, e.g., Diagnosis-Deceased, Progressive-Deceased.
+This pipeline investigates VAFs and TMB across matched samples. It produces pdf files containing the correlation of VAFs (with label genes of interest), as well as violin plots and stacked barplots for TMB of paired longitudinal samples: Diagnosis-Deceased, Progressive-Deceased, Recurrence-Deceased, Diagnosis-Progressive-Deceased, Diagnosis-Recurrence-Deceased, and Progressive-Recurrence-Deceased ...
```
```suggestion
This pipeline investigates VAFs and TMB across matched samples. It produces pdf files containing the correlation of VAFs (with labeled genes of interest), as well as violin plots and stacked barplots for TMB of paired longitudinal samples: Diagnosis-Deceased, Progressive-Deceased, Recurrence-Deceased, Diagnosis-Progressive-Deceased, Diagnosis-Recurrence-Deceased, and Progressive-Recurrence-Deceased.
```

rjcorb commented on analyses/tmb-vaf-longitudinal/README.md:
```diff
@@ -28,9 +28,25 @@ Then, we will leverage this information to create one column for the constant va
+
+Moreover, we explored TMB and the number of mutations across longitudinal samples with genomic assays. 
+
+Generally, Hypermutant TMB defined as ≤10 Mb, and Ultrahypermutant TMB defined as >100 mutations/Mb (https://pubmed.ncbi.nlm.nih.gov/29056344/). There are hyper mutant samples in the PBTA cohort. We excluded samples with >= 10 from downstream analysis. Attention is needed in cases with high number of mutations in only one timepoint as this will lead to un-matched longitudinal samples. We also rem ...
```
```suggestion
Generally, Hypermutant TMB defined as >=10 Mb, and Ultrahypermutant TMB defined as >100 mutations/Mb (https://pubmed.ncbi.nlm.nih.gov/29056344/). There are hyper mutant samples in the PBTA cohort. We excluded samples with >= 10 from downstream analysis. Attention is needed in cases with high number of mutations in only one timepoint as this will lead to un-matched longitudinal samples. We also remove those so we always have matched longitudinal samples.
```

rjcorb commented on analyses/tmb-vaf-longitudinal/03-explore-TMB-mut.Rmd:
```diff
@@ -0,0 +1,221 @@
+    print(fname)
+    
+    # Run function
+    p <- create_dumbbell_ct(tmb_df = df_ct_sub, 
+                                 ylim = ylim, 
+                                 ct_id = cancer_groups[i])
```
It looks like there are some patients with only one time point, but maybe the timepoints just have similar TMBs? In the High-grade glioma plot, for example.

AntoniaChroni replied:
@rjcorb I am not seeing this in the HGG on my end.

![Screenshot 2023-10-03 at 6 49 40 PM](https://github.com/d3b-center/pbta-tumor-evolution/assets/46328680/c63957a0-1681-4cd2-9e2b-c17dde54eb16)

But I can see this for the `Low-grade glioma,Glial-neuronal tumor-TMB-dumbbell.pdf`.  Here, I confirm that both timepoints have the same TMB value.
![Screenshot 2023-10-03 at 6 50 15 PM](https://github.com/d3b-center/pbta-tumor-evolution/assets/46328680/7b9f7d6f-861f-4193-b69f-ff0cde30b90c)

rjcorb reviewed (approved):
This looks great! Nice job with the plotting functions. I will second Ammar's comments that the text size could be larger in `TMB-genomic-no-hypermutants.pdf`. But I don't need to review again and will approve.

Commit:
Remove scratch dir, add results_dir and replace with tmb_vaf_genomic.tsv

Commit:
Merge branch 'main' into achroni/explore-TMB-mut-script

AntoniaChroni committed:
Update analyses/tmb-vaf-longitudinal/README.md

Co-authored-by: Ryan Corbett <corbettr@chop.edu>

AntoniaChroni committed:
Update analyses/tmb-vaf-longitudinal/README.md

Co-authored-by: Ryan Corbett <corbettr@chop.edu>

Commit:
Merge branch 'main' into achroni/explore-TMB-mut-script

Commit:
Merge remote-tracking branch 'origin/achroni/explore-TMB-mut-script' into achroni/explore-TMB-mut-script

Commit:
Change file size to avoid sample labels squished for TMB-genomic-total.pdf

Commit:
Add library(scales)

Commit:
Merge branch 'achroni/explore-TMB-mut-functions' into achroni/explore-TMB-mut-script

Commit:
Merge branch 'achroni/explore-TMB-mut-functions' into achroni/explore-TMB-mut-script

Commit:
Run in Docker with new code

Commit:
Merge branch 'achroni/explore-TMB-mut-functions' into achroni/explore-TMB-mut-script

Commit:
Merge branch 'achroni/explore-TMB-mut-functions' into achroni/explore-TMB-mut-script

Commit:
Remove old plots

Commit:
Edit size of figure in create-stacked-barplot

Commit:
Add cg_kids_id col and parameter

Commit:
Define and order palette in notebook

Commit:
Add palette as parameter in all functions

Commit:
Merge branch 'achroni/explore-TMB-mut-functions' into achroni/explore-TMB-mut-script

Commit:
Run in Docker after revising code

AntoniaChroni commented:
> The script logic makes sense to me and runs as expected. I do have some general suggestions, more related to plots:
> 
> `TMB-genomic-no-hypermutants.pdf`: Text size for axis and legend titles can be larger `TMB-genomic-total.pdf`: The sample labels are squished together and could be spaced out better. Also you can increase y-axis tick breaks (we can't really make out the smaller/ anything < 100 TMB).

@naqvia I have addressed these issues. A couple more changes after @rjcorb suggestions:
- passing `palette` as a parameter in functions
- `TMB-genomic-no-hypermutants.pdf` shows kids_id together with cancer group.

Commit:
Merge branch 'main' into achroni/genomic-paired

Commit:
Merge branch 'main' into achroni/explore-TMB-mut-script

Commit:
Merge branch 'main' into achroni/explore-TMB-mut-script

Commit:
Merge remote-tracking branch 'origin/main' into main

Commit:
Merge branch 'main' into achroni/explore-TMB-mut-script

naqvia reviewed (approved):
looks good to me!
